src/classes/UTIL_ACCT_Naming.cls

```diff
@@ -221,53 +221,86 @@ public class UTIL_ACCT_Naming {
             }
             listNameStrings.add(' ');
         }
+        
+        //Build nameContactFieldsString
+        //If setContactFields has lastname, we need to group contacts by lastname
         if (setContactFields.contains('lastname')) {
-
-            Integer lnCounter = mapLastNameContacts.keySet().size();
-
+            //Counter for last name loop
+            Integer lnCounter = 0;
             for (String lastName : mapLastNameContacts.keySet()) {
+                lnCounter++;
+                //Adding connector only if a last name has more than two contacts
+                //Starting at the second contact
+                if (mapLastNameContacts.keySet().size() > 1 && lnCounter > 1) {
+                    if (lnCounter != mapLastNameContacts.keySet().size()) {
+                        nameContactFieldsString += ', ';
+                    } else {
+                        nameContactFieldsString += Label.defaultNamingConnector + ' ';
+                    }
+                }
+
                 for (String nameString : listNameStrings) {
+                    String conNameString = '';
                     if (UTIL_Describe.isValidField('Contact', nameString.toLowerCase())) {
                         if (nameString == 'lastname') {
-                            result += lastname;
+                            conNameString += lastname;
                         } else {
-                            Integer counter = mapLastNameContacts.get(lastName).size();
+                            Integer conCounter = 0;
                             for (Contact con : mapLastNameContacts.get(lastName)) {
-                                result += con.get(nameString) != null ? con.get(nameString) : '';
-                                counter--;
-                                if (counter > 0) {
-                                    result += ' ' + Label.defaultNamingConnector + ' ';
+                                conCounter++;
+                                if (mapLastNameContacts.get(lastName).size() > 1 && conCounter > 1) {
+                                    if (conCounter != mapLastNameContacts.get(lastName).size()) {
+                                        conNameString += ', ';
+                                    } else {
+                                        conNameString += ' ' + Label.defaultNamingConnector + ' ';
+                                    }
+                                }
+                                if (con.get(nameString) != null) {
+                                    conNameString += con.get(nameString);
                                 }
                             }
                         }
+                        conNameString = conNameString.trim();
+                        if (conNameString != null
+                                && conNameString != '') {
+                            nameContactFieldsString += conNameString;
```

Comment:
Is this going to cause issues with spacing possibly? I did see that at one point in my testing that I wound up with Mr. and Mrs.Ryan and Carrie Family. So something isn't quite right here yet.